From 4fd7405823dd5dce823fd5710d1dd1a495091914 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 18 Jan 2019 13:44:21 +0200
Subject: [PATCH] [SW-0]: Makefile - rework install

Installation took too long so lets change a bit:
1) Remove unused files installation
2) Seperate into HOST (default) and LAB install targets
3) Introduce *-install-all target to install HOST & LAB
4) Rename GOYA_INSTALL_*_IP to HL_INSTALL_*_IP

Note: Update Jenkins jobs (otherwise item 4 will break jobs)

Change-Id: Ib81529e505ebbd9b0a651bd40481610d2ad1dcc6
---

diff --git a/Makefile b/Makefile
index e698eac..6a0124d 100644
--- a/Makefile
+++ b/Makefile
@@ -36,9 +36,9 @@
 	@echo '  optional environment variables:'
 	@echo '    PLDM_HEX_DIR       - Palladium directory to install flash image'
 	@echo '  goya-<board>-install - Install to <board> e.g. onix, pldm, asic'
-	@echo '                           set GOYA_INSTALL_LAB_IP=<lab IP> to'
+	@echo '                           set HL_INSTALL_LAB_IP=<lab IP> to'
 	@echo '                             override <board> default lab IP'
-	@echo '                           set GOYA_INSTALL_HOST_IP=<host IP> to'
+	@echo '                           set HL_INSTALL_HOST_IP=<host IP> to'
 	@echo '                             override <board> default host IP'
 	@echo
 	@echo 'Release:'
@@ -184,48 +184,52 @@
 IMAGES_DIR := ${OUTPUT_DIR}/images
 UBOOT_BASE := ${OUTPUT_DIR}/build/uboot-custom
 LINUX_BASE := ${OUTPUT_DIR}/build/linux-custom
-UBOOT_XMPL := ${UBOOT_BASE}/examples/standalone
 DSV_BASE := ${OUTPUT_DIR}/build/dsv-custom
-.PHONY: goya-onix-install goya-pldm-install goya-asic-install gaudi-onix-install gaudi-pldm-install
-goya-onix-install gaudi-onix-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
-goya-onix-install gaudi-onix-install: GOYA_INSTALL_HOST_IP ?= monster4
-goya-pldm-install gaudi-pldm-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
-goya-pldm-install gaudi-pldm-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.45
-goya-asic-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.105
-goya-asic-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.156
-gaudi-onix-install: gaudi-onix-target
-gaudi-pldm-install: gaudi-pldm-target
-goya-onix-install: goya-onix-target
+.PHONY: goya-onix-install goya-onix-install-all
+.PHONY: goya-pldm-install
+ PHONY: goya-asic-install goya-asic-install-all
+.PHONY: gaudi-onix-install gaudi-onix-install-all
+.PHONY: gaudi-pldm-install
+goya-onix-install gaudi-onix-install: HL_INSTALL_LAB_IP ?= monster4-nuc
+goya-onix-install gaudi-onix-install: HL_INSTALL_HOST_IP ?= monster4
+goya-pldm-install gaudi-pldm-install: HL_INSTALL_LAB_IP ?= 192.168.10.45
+goya-pldm-install gaudi-pldm-install: HL_INSTALL_HOST_IP ?= 192.168.10.45
+goya-asic-install gaudi-asic-install: HL_INSTALL_HOST_IP ?= 192.168.10.156
+goya-asic-install-all gaudi-asic-install-all: HL_INSTALL_HOST_IP ?= 192.168.10.156
+goya-asic-install-all gaudi-asic-install-all: HL_INSTALL_LAB_IP ?= 192.168.10.105
+
+goya-onix-install goya-onix-install-all: goya-onix-target
 goya-pldm-install: goya-pldm-target
-goya-asic-install: goya-asic-target
+goya-asic-install goya-asic-install-all: goya-asic-target
+gaudi-onix-install gaudi-onix-install-all: gaudi-onix-target
+gaudi-pldm-install: gaudi-pldm-target
+gaudi-asic-install gaudi-asic-install-all: gaudi-asic-target
 
-.PHONY: hl-install
-hl-install:
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(GOYA_INSTALL_HOST_IP):/usr/sbin/
-	-sshpass -p123456        scp $(IMAGES_DIR)/hl-smi			root@$(GOYA_INSTALL_HOST_IP):/usr/bin/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin	root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/
-	-sshpass -p123456        scp $(IMAGES_DIR)/boot.scr			root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot			root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp -r $(UBOOT_XMPL)/$(HL_DEV)*.bin        	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin			root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+.PHONY: hl-install-host hl-install-lab
+hl-install-host:
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(HL_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(HL_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(HL_INSTALL_HOST_IP):/usr/sbin/
+	-sshpass -p123456        scp $(IMAGES_DIR)/hl-smi			root@$(HL_INSTALL_HOST_IP):/usr/bin/
+hl-install-lab:
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin	root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot			root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 
 
-.PHONY: goya-install gaudi-install
-goya-install gaudi-install:
-	@$(MAKE) hl-install GOYA_INSTALL_HOST_IP=$(GOYA_INSTALL_HOST_IP) GOYA_INSTALL_LAB_IP=$(GOYA_INSTALL_LAB_IP)
+.PHONY: goya-install-host goya-install-lab
+.PHONY: gaudi-install-host gaudi-install-lab
+goya-install-host gaudi-install-host:
+	@$(MAKE) hl-install-host HL_INSTALL_HOST_IP=$(HL_INSTALL_HOST_IP)
+goya-install-lab gaudi-install-lab:
+	@$(MAKE) hl-install-lab HL_INSTALL_LAB_IP=$(HL_INSTALL_LAB_IP)
 
-gaudi-onix-install gaudi-pldm-install: gaudi-install
-goya-onix-install goya-pldm-install goya-asic-install: goya-install
-goya-onix-install gaudi-onix-install:
-	@
+goya-onix-install goya-pldm-install goya-asic-install: goya-install-host
+goya-onix-install-all goya-asic-install-all: goya-install-host goya-install-lab
+gaudi-onix-install gaudi-pldm-install gaudi-asic-install: gaudi-install-host
+gaudi-onix-install-all gaudi-asic-install-all: gaudi-install-host gaudi-install-lab
+
 PLDM_HEX_DIR ?= $(IMAGES_DIR)/tools/hl_shared_dir/pldm
 ${PLDM_HEX_DIR}:
 	@mkdir -p $@
@@ -233,7 +237,7 @@
 	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.hex $</
 goya-pldm-install gaudi-pldm-install:
 	@$(MAKE) $@-hex
-goya-asic-install:
+goya-asic-install goya-asic-install-all gaudi-asic-install gaudi-asic-install-all:
 	@
 
 .PHONY: goya-distclean gaudi-distclean
@@ -264,13 +268,13 @@
 	@HOST_UBOOT_TEST_HOOKS_BRD=$(patsubst $(HL_DEV)-%-test,$(HL_DEV)_%,$@)	\
 		$(MAKE) host-uboot-test-hooks-rebuild
 
-goya-onix-tools-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
-gaudi-onix-tools-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
+goya-onix-tools-install: HL_INSTALL_LAB_IP ?= monster4-nuc
+gaudi-onix-tools-install: HL_INSTALL_LAB_IP ?= monster4-nuc
 .PHONY: goya-onix-tools-install gaudi-onix-tools-install
 goya-onix-tools-install gaudi-onix-tools-install:
 	-cd board/habanalabs/common/ &&					\
 	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
-	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	root@$(HL_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 
 HOST_TBOT_TC ?= tc_ub_help.py
 HOST_TBOT_VERBOSE ?= off
@@ -293,17 +297,17 @@
 	PCIE_HOST_TBOT_LAB_IP=$(PCIE_HOST_TBOT_IP)		\
 	HOST_TBOT_VERB=$(HOST_TBOT_VERBOSE)			\
 		$(MAKE) host-tbot-rebuild
-goya-pldm-tools-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
+goya-pldm-tools-install: HL_INSTALL_LAB_IP ?= 192.168.10.45
 .PHONY: goya-pldm-tools-install
 goya-pldm-tools-install:
 	-cd board/habanalabs/common/ &&				\
 	sshpass -p123456 scp -r remote_power ip_pdu		\
-	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-goya-asic-tools-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.105
+	root@$(HL_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+goya-asic-tools-install: HL_INSTALL_LAB_IP ?= 192.168.10.105
 .PHONY: goya-asic-tools-install
 goya-asic-tools-install:
 	-cd board/habanalabs/common/ &&					\
 	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
 	local_remote_common local_run remote_run			\
 	$(IMAGES_DIR)/hl-fw-loader					\
-	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	root@$(HL_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
